File: 01a_File_formats_bonanza/javascript/read_xml_file.js
import fs from 'fs';
import xm12js from 'xml2js';

const parser = new xm12js.Parser();

fs.readFile('../me.xml', (err, data) => {
    if (err) {
        console.error('Error reading the file:', err);
        return;
    }
    
    parser.parseString(data, (err, result) => {
        if (err) {
            console.error('Error parsing XML:', err);
            return;
        }
        
        console.log(result);
        console.log('Name:', result.me.name[0]);
        console.log('Age:', result.me.age[0]);
        console.log('Hobbies:', result.me.hobbies[0].hobby);
    });
});

File: 01a_File_formats_bonanza/python/read_xml_file.py
import xml.etree.ElementTree as ET

tree = ET.parse('../me.xml')
root = tree.getroot()

name = root.find('name').text
age = root.find('age').text
hobbies = [hobby.text for hobby in root.find('hobbies').findall('hobby')]

print(f'Name: {name}')
print(f'Age: {age}')
print(f'Hobbies: {", ".join(hobbies)}')